pkgs/applications/networking/ssb/patchwork/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, appimage-run, fetchurl, runtimeShell, gsettings-desktop-schemas, gtk3, gobject-introspection, wrapGAppsHook }:
+
+let
+  # latest version that runs without errors
+  # https://github.com/ssbc/patchwork/issues/972
+  version = "3.11.4";
+in
+
+stdenv.mkDerivation rec {
+  name = "patchwork-${version}";
+
+  src = fetchurl {
+    url = "https://github.com/ssbc/patchwork/releases/download/v${version}/Patchwork-${version}-linux-x86_64.AppImage";
+    sha256 = "1blsprpkvm0ws9b96gb36f0rbf8f5jgmw4x6dsb1kswr4ysf591s";
+  };
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+  buildInputs = [ appimage-run gtk3 gsettings-desktop-schemas gobject-introspection ];
+
+  unpackPhase = ":";
+
+  installPhase = ''
+    mkdir -p $out/{bin,share}
+    cp $src $out/share/patchwork.AppImage
+    echo "#!${runtimeShell}" > $out/bin/patchwork
+    echo "${appimage-run}/bin/appimage-run $out/share/patchwork.AppImage" >> $out/bin/patchwork
+    chmod +x $out/bin/patchwork $out/share/patchwork.AppImage
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A decentralized messaging and sharing app built on top of Secure Scuttlebutt (SSB).";
+    longDescription = ''
+      sea-slang for gossip - a scuttlebutt is basically a watercooler on a ship.
+    '';
+    homepage = https://www.scuttlebutt.nz/;
+    license = licenses.agpl3;
+    maintainers = with maintainers; [ thedavidmeister ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
How come only `x86_64-linux` and not e.g. `platforms.linux` or `platforms.unix`?

Reply:
@asymmetric i don't know, i copied this from elsewhere `platforms.linux` would make sense i think

Reply:
I would try with the most general possible and see if Hydra/ofBorg tests fail, and then shrink the scope if they do.

Not sure this is the correct approach and maybe `with platforms; [ linux darwin ]` is enough.

Reply:
Looking at https://github.com/ssbc/patchwork/releases/, upstream only publishes x86_64 appimages. So if we don't compile from source, this will only work on "x86_64-linux".

Reply:
Of course - AppImage is Linux only 🤦‍♂

Reply:
i did try to do node2nix and similar, and asked on #nixos IRC a few times for help, and someone else in ssb did try as well..

it's too hard for beginners to get node working

can we get the AppImage in for linux and then if someone more knowledgeable wants to get the electron app building cross-platform they can?